Tests/AmplitudeTests.m

```diff
@@ -637,123 +615,122 @@ -(void)testSetOfflineTruncate {
-    XCTAssertEqual([dbHelper getTotalEventCount], 0);
 }
 
 -(void)testTruncateEventsQueues {
     int eventMaxCount = 50;
+
     XCTAssertGreaterThanOrEqual(eventMaxCount, kAMPEventRemoveBatchSize);
     self.amplitude.eventMaxCount = eventMaxCount;
 
-    AMPDatabaseHelper *dbHelper = [AMPDatabaseHelper getDatabaseHelper];
-    [self.amplitude setOffline:YES];
     for (int i = 0; i < eventMaxCount; i++) {
         [self.amplitude logEvent:@"test"];
     }
     [self.amplitude flushQueue];
-    XCTAssertEqual([dbHelper getEventCount], eventMaxCount);
+    XCTAssertEqual([self.amplitude.eventsBuffer count], eventMaxCount);
 
     [self.amplitude logEvent:@"test"];
     [self.amplitude flushQueue];
-    XCTAssertEqual([dbHelper getEventCount], eventMaxCount - (eventMaxCount/10) + 1);
+    XCTAssertEqual([self.amplitude.eventsBuffer count], eventMaxCount - (eventMaxCount/10) + 1);
 }
 
 -(void)testTruncateEventsQueuesWithOneEvent {
+    [self.amplitude setEventUploadThreshold:1];
     int eventMaxCount = 1;
     self.amplitude.eventMaxCount = eventMaxCount;
 
-    AMPDatabaseHelper *dbHelper = [AMPDatabaseHelper getDatabaseHelper];
     [self.amplitude logEvent:@"test1"];
     [self.amplitude flushQueue];
-    XCTAssertEqual([dbHelper getEventCount], eventMaxCount);
+    XCTAssertEqual([self.amplitude getEventCount], eventMaxCount);
 
+    [self.amplitude cleanUp];
+    self.amplitude.eventsBuffer = [[NSMutableArray alloc] init];
+    [self.amplitude setEventUploadThreshold:1];
+    self.amplitude.updatingCurrently = NO;
+    
     [self.amplitude logEvent:@"test2"];
     [self.amplitude flushQueue];
-    XCTAssertEqual([dbHelper getEventCount], eventMaxCount);
+    XCTAssertEqual([self.amplitude getEventCount], eventMaxCount);
 
     NSDictionary *event = [self.amplitude getLastEvent];
     XCTAssertEqualObjects([event objectForKey:@"event_type"], @"test2");
 }
 
 -(void)testInvalidJSONEventProperties {
+    [self.amplitude setEventUploadThreshold:1];
+    
     NSURL *url = [NSURL URLWithString:@"https://amplitude.com/"];
     NSDictionary *properties = [NSDictionary dictionaryWithObjectsAndKeys:url, url, url, @"url", nil];
     [self.amplitude logEvent:@"test" withEventProperties:properties];
     [self.amplitude flushQueue];
-    XCTAssertEqual([[AMPDatabaseHelper getDatabaseHelper] getEventCount], 1);
+    XCTAssertEqual([self.amplitude getEventCount], 1);
 }
 
 -(void)testClearUserProperties {
-    AMPDatabaseHelper *dbHelper = [AMPDatabaseHelper getDatabaseHelper];
     [self.amplitude setEventUploadThreshold:2];
 
     [self.amplitude clearUserProperties];
     [self.amplitude flushQueue];
 
-    XCTAssertEqual([dbHelper getEventCount], 0);
-    XCTAssertEqual([dbHelper getIdentifyCount], 1);
-    XCTAssertEqual([dbHelper getTotalEventCount], 1);
+    XCTAssertEqual([self.amplitude getEventCount], 0);
+    XCTAssertEqual([self.amplitude getIdentifyCount], 0);
+    XCTAssertEqual([self.amplitude.identifyBuffer count], 1);
```

Comment:
Not based on the previous result. Because there are no log events, it's will not store events in the file storage because of https://github.com/amplitude/Amplitude-iOS/blob/main/Sources/Amplitude/Amplitude.m#L748.
So the only way we can test is if the events are stored in the buffer.